elasticsearch/helpers/actions.py

```diff
@@ -231,10 +254,48 @@ def _chunk_actions(
     chunker = _ActionChunker(
         chunk_size=chunk_size, max_chunk_bytes=max_chunk_bytes, serializer=serializer
     )
-    for action, data in actions:
-        ret = chunker.feed(action, data)
-        if ret:
-            yield ret
+
+    if not flush_after_seconds:
+        for action, data in actions:
+            ret = chunker.feed(action, data)
+            if ret:
+                yield ret
+    else:
```

Comment:
If there is a timeout, then we put the for-loop that iterates over the data elements in a separate task that puts the items in a queue. And here we wait on the queue with a timeout, so that we can insert a flush when data does not come in a while.